airflow/executors/celery_executor.py

```diff
@@ -547,11 +547,12 @@ def get_many(self, async_results) -> Mapping[str, EventBufferValueType]:
         """Gets status for many Celery tasks using the best method available."""
         if isinstance(app.backend, BaseKeyValueStoreBackend):
             result = self._get_many_from_kv_backend(async_results)
-            return result
-        if isinstance(app.backend, DatabaseBackend):
+        elif isinstance(app.backend, DatabaseBackend):
             result = self._get_many_from_db_backend(async_results)
-            return result
-        result = self._get_many_using_multiprocessing(async_results)
+        else:
+            async_results = list(async_results)
+            result = self._get_many_using_multiprocessing(async_results)
+        async_results = list(async_results) if isinstance(async_results, map) else async_results
```

Comment:
can you please add a test -- similar to the one you mentioned in the PR description

Reply:
@kaxil Done. One of these days I'll have a PR that includes a test ;)

Reply:
@kaxil Looks like the failed build is due to a bug that was corrected yesterday #14986